src/custom/state/orders/updaters/APIOrdersUpdater.ts

```diff
@@ -0,0 +1,116 @@
+import { useAllTokens } from 'hooks/Tokens'
+import { Order, OrderStatus } from 'state/orders/actions'
+import { NATIVE_CURRENCY_BUY_ADDRESS, NATIVE_CURRENCY_BUY_TOKEN } from 'constants/index'
+import { ChainId } from 'state/lists/actions'
+import { ApiOrderStatus, classifyOrder } from 'state/orders/utils'
+import { computeOrderSummary } from 'state/orders/updaters/utils'
+
+function getToken(address: string, chainId: ChainId, tokens: { [p: string]: Token }): Token | undefined {
+  if (address.toLowerCase() === NATIVE_CURRENCY_BUY_ADDRESS.toLowerCase()) {
+    return NATIVE_CURRENCY_BUY_TOKEN[chainId]
+  }
+  return tokens[getAddress(address)]
+}
+
+function classifyLocalStatus(status: ApiOrderStatus): OrderStatus | undefined {
+  switch (status) {
+    case 'cancelled':
+      return OrderStatus.CANCELLED
+    case 'expired':
+      return OrderStatus.EXPIRED
+    case 'pending':
+      return OrderStatus.PENDING
+    case 'fulfilled':
+      return OrderStatus.FULFILLED
+    default:
+      return undefined
+  }
+}
+
+function transformApiOrderToStoreOrder(
+  order: OrderMetaData,
+  chainId: ChainId,
+  allTokens: { [address: string]: Token }
+): Order | undefined {
+  const { uid: id, sellToken, buyToken, creationDate: creationTime, receiver } = order
+
+  // TODO: load tokens from network when not found on local state
+  const inputToken = getToken(sellToken, chainId, allTokens)
+  const outputToken = getToken(buyToken, chainId, allTokens)
+
+  const apiStatus = classifyOrder(order)
+  const status = classifyLocalStatus(apiStatus)
+
+  if (!status) {
+    console.warn(`APIOrdersUpdater::Order ${id} in unknown internal state: ${apiStatus}`)
+    return
+  }
+
+  if (inputToken && outputToken) {
+    const storeOrder: Order = {
+      ...order,
+      inputToken,
+      outputToken,
+      id,
+      creationTime,
+      summary: '',
+      status,
+      receiver,
+      apiAdditionalInfo: order,
+      isCancelling: apiStatus === 'pending' && order.invalidated, // already cancelled in the API, not yet in the UI
+    }
+    // The function to compute the summary needs the Order instance to exist already
+    // That's why it's not used before and an empty string is set instead
+    storeOrder.summary = computeOrderSummary({ orderFromStore: storeOrder, orderFromApi: order }) || ''
+
+    return storeOrder
+  } else {
+    console.warn(
+      `APIOrdersUpdater::Tokens not found for order ${id}: sellToken ${!inputToken ? sellToken : 'found'} - buyToken ${
+        !outputToken ? buyToken : 'found'
+      }`
+    )
+    return
+  }
+}
+
+export function APIOrdersUpdater(): null {
+  const { account, chainId } = useActiveWeb3React()
+  const allTokens = useAllTokens()
+  const tokenAreLoaded = useMemo(() => Object.keys(allTokens).length > 0, [allTokens])
+  const addOrUpdateOrdersBatch = useAddOrUpdateOrdersBatch()
+
+  useEffect(() => {
+    if (account && chainId && tokenAreLoaded) {
+      getOrders(chainId, account, 100)
```

Comment:
I've explained on https://github.com/gnosis/cowswap/pull/1395//#discussion_r722421384

Short version: there's no limit for pending orders, so I load far enough that should account for all pending orders

When there's such a filter on the backend this can be revisited.

And I'll remove the magic number